Roommatch/Roommatch/ViewControllers/LoginViewController.m

```diff
@@ -0,0 +1,89 @@
+//
+//  LoginViewController.m
+//  Roommatch
+//
+//  Created by Lily Yang on 7/7/22.
+//
+
+#import "LoginViewController.h"
+#import "User.h"
+
+@interface LoginViewController ()
+@property (weak, nonatomic) IBOutlet UITextField *usernameTextField;
+@property (weak, nonatomic) IBOutlet UITextField *passwordTextField;
+
+@end
+
+@implementation LoginViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    // Do any additional setup after loading the view.
+}
+
+- (IBAction)signUp:(id)sender {
+    NSString *username = self.usernameTextField.text;
+    NSString *password = self.passwordTextField.text;
+
+    if([username isEqualToString:@""] || [password isEqualToString:@""]){
+        [self alertNoUsernameOrPassword];
+        return;
+    }
+    
+    User *user = [User user];
+    user.username = username;
+    user.password = password;
+    
+    [user signUpInBackgroundWithBlock:^(BOOL succeeded, NSError *error) {
+        if (error != nil) {
+            NSLog(@"Error: %@", error.localizedDescription);
+        } else {
+            NSLog(@"User registered successfully");
+            [self performSegueWithIdentifier:@"Sign Up Segue" sender:nil];
```

Comment:
It could be good to use a weak reference to `self` here since you're making an asynchronous call; [example](https://www.internalfb.com/code/fbsource/[c9425a1fff37]/fbobjc/Apps/Diamond/Viewer/Viewer/DVAppDelegate.mm?lines=330%2C335). That way it doesn't create a reference cycle.